.tekton/tasks/upload-currency-report.yaml

```diff
@@ -0,0 +1,54 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: upload-currency-report
+spec:
+  params:
+    - name: continuous-delivery-context-secret
+      default: secure-properties
+    - name: type
+      value: $(params.type)
+    - name: node-version
+      value: $(params.node-version)
+  workspaces:
+    - name: output
+      mountPath: /artifacts
+  steps:
+    - name: execute
+      image: node:$(params.node-version)
+      env:
+        - name: GH_ENTERPRISE_TOKEN
+          valueFrom:
+            secretKeyRef:
+              name: $(params.continuous-delivery-context-secret)
+              key: "GH_ENTERPRISE_TOKEN"
+      script: |
+        #!/bin/bash
+
+        if [ "$(params.type)" != "main" ]; then
```

Comment:
FYI: We upload the report only for the main pipeline. Nightly and weekly jobs are excluded. I wanted to avoid adding a new pipeline only for the report. We can change that in the future if we want